src/main/java/io/kestra/plugin/nats/Deserializer.java

```diff
@@ -0,0 +1,19 @@
+package io.kestra.plugin.nats;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.AllArgsConstructor;
+
+import java.util.function.Function;
+
+@AllArgsConstructor
+@Schema(title = "Deserializer to use when retrieving NATS message data", description = "Data is in bytes format by default")
+public enum Deserializer {
```

Comment:
I understand why you did this but I'm not a fan.
For the PubSub Consume task, I base64 encode the message.

I don't think byte[] has any use in Kestra, and for a useful deserializer it would be best to support JSON ... but then not everybody uses JSON.

So I think for now the best is to base64encore the message and store it like this, you can check what I did with the PubSub Consume task. Users can use Pebble functions to base64 decode, convert to JSON, or everything else

Reply:
Sure :)